required_for_pub.module

```diff
@@ -194,136 +182,122 @@ function required_for_pub_form_validate($form, &$form_state) {
-    $func = $module.'_field_is_empty';
-    
-    // special handling for booleans
-    // NOTE: the list.module's hook_field_is_empty specifies this: 
-    // if (empty($item['value']) && (string) $item['value'] !== '0') { return TRUE; }
-    // paramter $item passed is array('value' => 0) -- if user has not clicked the wdiget form
-    // which means that the boolean field will always return hook_field_is_empty to be FALSE -- that is, the field contains data. when in fact, it should be treated as EMPTY.
-    // hence, we convert the value from 0 to NULL or set $item to NULL
-    if ($field_type == 'list_boolean') {
-      if ($item['value'] == 0) {
-        $item = NULL;
-      }
-    }
-    
-    // special handling for fields that do no use delta or the 0 offset.
-    if (empty($item)) {
-      $field_is_empty = TRUE;
-    }
-    else { 
-      $field_is_empty = call_user_func($func, $item, $field);
-    }
-    
-    // TO CHECK if field is empty -- this will account for all field types
-    if ($field_is_empty) {
+  foreach ($form_state['required_for_pub']['#fields'] as $field_name) {
+    if (_required_for_pub_field_is_empty($form_state, $field_name, $lang)) {
       $empty_fields[] = $field_name;
     }
   }
   
   // the most important validation
-  $field_string = "";
   $digit = 1;
-  if ($status == 1 && !empty($empty_fields) ) {
-    // the form is cached. Hence, the css file must be loaded again in the validate function  or  the css wonk work.
+  if (!empty($empty_fields) ) {
+    // The form is cached. Hence, the css file must be loaded again in the
+    // validate function or the css won't work.
     drupal_add_css(drupal_get_path('module', 'required_for_pub') . '/css/required_for_pub.css', array('group' => CSS_DEFAULT, 'every_page' => TRUE));
     if (module_exists('save_draft')) {
       form_set_error('required_for_pub', t('You must do "a" or "b": <p>a. Save this content as DRAFT</p> <p>b. Or PULBISH it but FILL UP "required fields for content to be published" listed below:</p>'));       
     }
     else {
-    
-      form_set_error('required_for_pub', t('You must do "a" or "b": <p>a. Save this content as DRAFT.</p> <p>b. Or PUBLISH it but FILL UP the "required fields for content to be published" listed below:</p>'));       
+      form_set_error('required_for_pub', t('You must do "a" or "b": <p>a. Save this content as DRAFT.</p> <p>b. Or PUBLISH it but FILL UP the "required fields for content to be published" listed below:</p>'));
     }
 
     foreach ($empty_fields as $key => $field_name) {
-      $field_string = $field_name;
-      
       $label = $form_state['required_for_pub']['#fields_instances'][$field_name]['label'];
       form_set_error($field_name, t('%digit. %field_name', array('%field_name' => $label, '%digit' => $digit)));
       $digit = $digit + 1;
     }
-    
     return FALSE;
-    
   }
   else {
```

Comment:
I wonder if you still need this else or can just return TRUE at the end as a fallback